File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    res = 0

    for i in range(1, number + 1):
        cnt = 0
        for j in range(1, int(i**0.5) + 1):
            if i % j == 0:
                if i // j == j:
                    cnt += 1
                else:
                    cnt += 2
            if cnt > limit:
                cnt = power
                break
        res += cnt
    return res

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
  let res = 0
  
  for (let i = 1; i <= number; i++) {
    let cnt = 0

    for (let j = 1; j <= Math.sqrt(i); j++) {
      if (i % j === 0) {
        if (i / j === j) cnt += 1
        else cnt += 2
      }
      if (cnt > limit) {
        cnt = power
        break
      }
    }
    res += cnt
  }
  return res
}